pkg/api/v1/atlasdatabaseuser_types.go

```diff
@@ -52,11 +52,14 @@ const (
 )
 
 // AtlasDatabaseUserSpec defines the desired state of Database User in Atlas
+// +kubebuilder:validation:XValidation:rule="has(self.atlasRef) && !has(self.projectRef) ? true : !has(self.atlasRef) && has(self.projectRef)",message="cannot define both atlasRef and projectRef at same time"
 type AtlasDatabaseUserSpec struct {
 	api.LocalCredentialHolder `json:",inline"`
 
 	// Project is a reference to AtlasProject resource the user belongs to
-	Project common.ResourceRefNamespaced `json:"projectRef"`
+	Project *common.ResourceRefNamespaced `json:"projectRef,omitempty"`
+	// AtlasRef holds the Atlas project ID the user belongs to
+	AtlasRef *ExternalProjectReference `json:"atlasRef,omitempty"`
```

Comment:
The name `atlasRef` is ambiguous, it does not indicate what we are pointing to (only the documentation reveals this is pointing to projects).

Suggestion for a name: `externalProjectRef` (this makes it a nice symmetry to `projectRef`). This would translate into the following yaml:

```
kind: AtlasDatabaseUser
spec:
  externalProjectRef:
    id: 123-456-789
```
vs.
```
kind: AtlasDatabaseUser
spec:
  projectRef:
    name: foo
    namespace: bar
```

Reply:
Agreed.